src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Specs/Types/CollectionSpec.cs

```diff
@@ -23,6 +23,8 @@ internal abstract record CollectionWithCtorInitSpec : CollectionSpec
         public required CollectionInstantiationConcreteType InstantiationConcreteType { get; init; }
 
         public required CollectionPopulationCastType PopulationCastType { get; init; }
+
+        public required bool ShouldTestCast {get; init; }
```

Comment:
(nit)
```suggestion
        public required bool ShouldTestCast { get; init; }
```

Reply:
Would it be better to name this `ShouldTryCast`? Or invert it to `UseDirectCast`?

I'm not sure I understood what "test cast" meant right away.

Reply:
Yeah, the naming is hard.  I think `ShouldTryCast` is better.

Reply:
FWIW it is called `test` in the C# language https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast

Reply:
Yeah - it might just be me then. I guess I hadn't heard that phrase used - but I'm not really a language expert.

If you want to name it back, I wouldn't object.